### ISSUE TITLE
Fixed Fan-in and custom model

### ISSUE THREAD

**Issue by SampadChowdhury** [calgaryml/condensed-sparsity#78]
Hi,
First of all great work, and thankyou for making it public.
I haven't yet delved deeper into the implementation, but I wanted to know that I have a network which is rather simple- a 2 layer of MLP neural network, but has some modifications- instead of using relu/sigmoid, i have custom sigmoid, which gets steeper as training progresses, and at inference it acts like a heavside function (in short, the activations are binary during inference).
I wanted to get some insights about if the framework you have might work with this type of NN?

Also what I understood for the paper is that there is neuron ablation happeining, and a result of it is the weight connections gets redistributed to other neurons. I have a constraint that at the end of my training (and inference), i want my neurons to retain specified fan-in ('n' connections/neuron), where i already know **'n'** before hand. Is there any provision to do that? Declare my **'n'** before hand, and let the algorithm do its work?  Is there any repurcations i should be aware of?

I would like some insights from you on this, thankyou!!
